core/stack.go

```diff
@@ -40,6 +43,11 @@ func CreateStack(cfg *Config) (*stack.Stack, error) {
 		opts = append(opts, cfg.Options...)
 	}
 
+	tcpSockOpts := append(
+		option.DefaultTCPSocketOptions(),
+		cfg.TCPSocketOptions...,
```

Comment:
I'm a bit concerned about the potential for duplicate calls; the current way to call this option chain might not be ideal/efficient because they might be called for the same option with different values for every TCP endpoint.

I'm not sure what the best way to do it is, but I'd prefer to avoid calling an option multiple times.